code/modules/library/lib_machines.dm

```diff
@@ -104,6 +105,9 @@
 	popup.open()
 
 /obj/machinery/computer/libraryconsole/Topic(href, href_list)
+	if(world.time < topic_timeout + 10)
+		return

```

Comment:
I thought we recently added a helper for this @Rohesie @ExcessiveUseOfCobblestone @ninjanomnom, can't remember what it was

Reply:
@AnturK

Reply:
COOLDOWN_ defines yes.

Reply:
lol yes

Reply:
This will rarely get used, so define a new cooldown index for topics in general and just do something like this:
```nix
/obj/machinery/computer/libraryconsole/Topic(href, href_list)
	if(TIMER_COOLDOWN_CHECK(usr, COOLDOWN_TOPIC))
		return
	TIMER_COOLDOWN_START(usr, COOLDOWN_TOPIC, 1 SECONDS)
```

If you want to generalize it for all topics, then add the cooldown variable to the base of mob:
`COOLDOWN_DECLARE(topics_cooldown)` (or just add the variable)
And then instead just do something like this:
```nix
/obj/machinery/computer/libraryconsole/Topic(href, href_list)
	if(!COOLDOWN_FINISHED(usr, topics_cooldown))
		return
	COOLDOWN_START(usr, topics_cooldown, 1 SECONDS)
```

Reply:
we already have a TOPIC cooldown